Autoscend aborting out on Normal Palindome because it's out of pulls for Wet Stew

# Context

* KolMafia version: r28024
* Class: Seal Clubber
* Run type (Normal):
* Challenge Path (Standard):
* Number of Ascensions: 483

# Expected/Desired Behavior
If unable to complete a portion of a quest, and with other doable quests remaining, Autoscend should move on to the next quest.

# Current Behavior
When autoscend gets to task 50 (L11_palindome) and is in normal/softcore and lacks available pulls for Wet Stew, rather than bypassing task 50 and moving on to other ones like it does with other tasks (i.e. bypassing filthworms when you are afflicted with ELY), it aborts out with a generic error:
```
[ERROR] Palindome failure:
[ERROR] You probably just need to get a Mega Gem to fix this.
We have made too much progress in the Palindome and should not be here.
```

## Failure Logs
```
[INFO] Turn(329): Starting with 5 left and 0 pulls left at Level: 13
[INFO] Encounter: -25.0 Exp Bonus: 36.244
[INFO] Meat Drop: 185.0 Item Drop: 85.0
[INFO] HP: 882/882, MP: 339/534, Meat: 73916
[INFO] Tummy: 15/15 Liver: 14/14 Spleen: 0/15
[INFO] ML: 118 control: 0
[INFO] Everything Looks Yellow: 95
[INFO] Delay between adventures... beep boop... 
Countdown: 1 second...
Waiting completed.
Preference _auto_thisLoopHandleFamiliar changed from true to false
Preference auto_familiarChoice changed from Adorable Seal Larva to
Preference _auto_tunedElement changed from cold to
[DEBUG] Resetting auto_maximize_current to 5item,meat,0.5initiative,0.1da 1000max,dr,0.5all res,1.5mainstat,-fumble,mox,0.4hp,0.2mp 1000max,3mp regen,1.5weapon damage,0.75weapon damage percent,1.5elemental damage,2familiar weight,-equip little bitty bathysphere
Preference _auto_maximize_equip_acc1 changed from spring shoes to
[DEBUG] Attempting to execute task 0 LX_freeCombatsTask
[DEBUG] Attempting to execute task 1 woods_questStart
[DEBUG] Attempting to execute task 2 LX_unlockPirateRealm
[DEBUG] Attempting to execute task 3 catBurglarHeist
[DEBUG] Attempting to execute task 4 auto_breakfastCounterVisit
[DEBUG] Attempting to execute task 5 chateauPainting
[DEBUG] Attempting to execute task 6 LX_setWorkshed
[WARNING] You have no workshed to change to so leaving it as model train set
[DEBUG] Attempting to execute task 7 LX_galaktikSubQuest
[DEBUG] Attempting to execute task 8 L9_leafletQuest
[DEBUG] Attempting to execute task 9 L5_findKnob
[DEBUG] Attempting to execute task 10 L12_sonofaPrefix
[DEBUG] Attempting to execute task 11 LX_burnDelay
[DEBUG] Attempting to execute task 12 LX_summonMonster
[DEBUG] Attempting to execute task 13 LM_edTheUndying
[DEBUG] Attempting to execute task 14 LX_bugbearInvasion
[DEBUG] Attempting to execute task 15 LX_lowkeySummer
[DEBUG] Attempting to execute task 16 L11_aridDesert
[DEBUG] Attempting to execute task 17 L11_shenStartQuest
[DEBUG] Attempting to execute task 18 finishBuildingSmutOrcBridge
[DEBUG] Attempting to execute task 19 LX_goingUnderground
[DEBUG] Attempting to execute task 20 LX_useBreathitinCharges
[DEBUG] Attempting to execute task 21 LX_guildUnlock
[DEBUG] Attempting to execute task 22 LX_unlockDesert
[DEBUG] Attempting to execute task 23 LX_lockPicking
[DEBUG] Attempting to execute task 24 LX_fatLootToken
[DEBUG] Attempting to execute task 25 LX_steelOrgan
[DEBUG] Attempting to execute task 26 LX_spookyravenManorFirstFloor
[DEBUG] Attempting to execute task 27 L5_getEncryptionKey
[DEBUG] Attempting to execute task 28 L5_findKnob
[DEBUG] Attempting to execute task 29 L12_islandWar
[INFO] Delaying filthworms because Everything Looks Yellow
[DEBUG] Attempting to execute task 30 L11_blackMarket
[DEBUG] Attempting to execute task 31 L11_forgedDocuments
[DEBUG] Attempting to execute task 32 L11_mcmuffinDiary
[DEBUG] Attempting to execute task 33 L11_getBeehive
[DEBUG] Attempting to execute task 34 L2_mosquito
[DEBUG] Attempting to execute task 35 LX_unlockHiddenTemple
[DEBUG] Attempting to execute task 36 L6_dakotaFanning
[DEBUG] Attempting to execute task 37 L11_unlockHiddenCity
[DEBUG] Attempting to execute task 38 L11_hiddenCityZones
[DEBUG] Attempting to execute task 39 L11_hiddenCity
[INFO] Target hp => 882 - Considering restore options at 882/882 HP with 339/534 MP
[INFO] Active Negative Effects => []
[INFO] The idden [sic] apartment!
[INFO] Hidden Apartment Progress: 0
[DEBUG] Don't know how to unlock The Hidden Apartment Building
[WARNING] Can't get to The Hidden Apartment Building right now.
[DEBUG] Attempting to execute task 40 LX_spookyravenManorSecondFloor
[INFO] Spookyraven: Bathroom
[DEBUG] Attempting to execute task 41 L11_mauriceSpookyraven
[DEBUG] Attempting to execute task 42 L11_talismanOfNam
[DEBUG] Attempting to execute task 43 L10_plantThatBean
[DEBUG] Attempting to execute task 44 L10_rainOnThePlains
[INFO] Castle (Top Floor) - Finishing L10 Quest.
[DEBUG] Attempting to execute task 45 L9_chasmBuild
[DEBUG] Attempting to execute task 46 L9_highLandlord
[DEBUG] Attempting to execute task 47 L8_trapperQuest
[DEBUG] Attempting to execute task 48 L7_crypt
[DEBUG] Attempting to execute task 49 L6_friarsGetParts
[DEBUG] Attempting to execute task 50 L11_palindome
[INFO] In the palindome : emodnilap eht nI
[ERROR] Palindome failure:
[ERROR] You probably just need to get a Mega Gem to fix this.
We have made too much progress in the Palindome and should not be here.
Stack trace:
at L11_palindome (level_11.ash:3074)
at process_tasks (autoscend.ash:1692)
at doTasks (autoscend.ash:1925)
at auto_begin (autoscend.ash:2041)
at safe_preference_reset_wrapper (autoscend.ash:2071)
at safe_preference_reset_wrapper (autoscend.ash:2078)
at safe_preference_reset_wrapper (autoscend.ash:2078)
at safe_preference_reset_wrapper (autoscend.ash:2078)
at main (autoscend.ash:2109)
```

[vuvuzela_virtuoso_20240824.txt](https://github.com/user-attachments/files/16737728/vuvuzela_virtuoso_20240824.txt)

--- a/RELEASE/scripts/autoscend.ash
+++ b/RELEASE/scripts/autoscend.ash
@@ -175,6 +175,7 @@ void initializeSettings() {
 	remove_property("auto_beatenUpLocations");
 	set_property("auto_getBeehive", false);
 	set_property("auto_bruteForcePalindome", false);
+	set_property("auto_doWhiteys", false);
 	set_property("auto_cabinetsencountered", 0);
 	set_property("auto_chasmBusted", true);
 	set_property("auto_chewed", "");

--- a/RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash
+++ b/RELEASE/scripts/autoscend/combat/auto_combat_default_stage2.ash
@@ -299,6 +299,11 @@ string auto_combatDefaultStage2(int round, monster enemy, string text)
 			couldInstaKill = false;
 		}
 	}
+	else if($monsters[Racecar Bob, Bob Racecar] contains enemy && item_amount($item[photograph of a dog]) == 0 && internalQuestStatus("questL11Palindome") < 2)
+	{
+		//don't want to instakill if we haven't used the disposable camera yet
+		couldInstaKill = false;
+	}
 	else if(wantToForceDrop(enemy))
 	{
 		//want drops from this enemy

--- a/RELEASE/scripts/autoscend/quests/level_11.ash
+++ b/RELEASE/scripts/autoscend/quests/level_11.ash
@@ -2924,13 +2924,57 @@ boolean L11_palindome()
 	#	In hardcore, guild-class, the right side of the or doesn't happen properly due us farming the
 	#	Mega Gem within the if, with pulls, it works fine. Need to fix this. This is bad.
 	#
-	if((item_amount($item[Bird Rib]) > 0) && (item_amount($item[Lion Oil]) > 0) && (item_amount($item[Wet Stew]) == 0))
+	boolean doWhiteys()
 	{
-		autoCraft("cook", 1, $item[Bird Rib], $item[Lion Oil]);
+		if(item_amount($item[white page]) > 0)
+		{
+			set_property("choiceAdventure940", 1);
+			if(item_amount($item[Bird Rib]) > 0)
+			{
+				set_property("choiceAdventure940", 2);
+			}
+
+			if(get_property("lastGuildStoreOpen").to_int() < my_ascensions())
+			{
+				auto_log_warning("This is probably no longer needed as of r16907. Please remove me", "blue");
+				auto_log_warning("Going to pretend we have unlocked the Guild because Mafia will assume we need to do that before going to Whitey's Grove and screw up us. We'll fix it afterwards.", "red");
+			}
+			backupSetting("lastGuildStoreOpen", my_ascensions());
+			string[int] pages;
+			pages[0] = "inv_use.php?pwd&which=3&whichitem=7555";
+			pages[1] = "choice.php?pwd&whichchoice=940&option=" + get_property("choiceAdventure940");
+			if(autoAdvBypass(0, pages, $location[Whitey\'s Grove], "")) {}
+			restoreSetting("lastGuildStoreOpen");
+			return true;
+		}
+		// +item is nice to get that food
+		bat_formBats();
+		auto_lostStomach(true);
+		auto_log_info("Off to the grove for some doofy food!", "blue");
+		return autoAdv(1, $location[Whitey\'s Grove]);
 	}
-	if((item_amount($item[Stunt Nuts]) > 0) && (item_amount($item[Wet Stew]) > 0) && (item_amount($item[Wet Stunt Nut Stew]) == 0))
+
+	boolean makeWetStuntNutStew()
 	{
-		autoCraft("cook", 1, $item[wet stew], $item[stunt nuts]);
+		 if((item_amount($item[Bird Rib]) > 0) && (item_amount($item[Lion Oil]) > 0) && (item_amount($item[Wet Stew]) == 0))
+		{
+			autoCraft("cook", 1, $item[Bird Rib], $item[Lion Oil]);
+		}
+
+		if((item_amount($item[Stunt Nuts]) > 0) && (item_amount($item[Wet Stew]) > 0) && (item_amount($item[Wet Stunt Nut Stew]) == 0))
+		{
+			autoCraft("cook", 1, $item[wet stew], $item[stunt nuts]);
+		}
+		if(item_amount($item[wet stunt nut stew]) > 0)
+		{
+			return true;
+		}
+		return false;
+	}
+
+	if(item_amount($item[wet stunt nut stew]) == 0 && internalQuestStatus("questL11Palindome") > 3)
+	{
+		return makeWetStuntNutStew();
 	}
 
 	if((item_amount($item[Wet Stunt Nut Stew]) > 0) && !possessEquipment($item[Mega Gem]))
@@ -2940,39 +2984,14 @@ boolean L11_palindome()
 		visit_url("place.php?whichplace=palindome&action=pal_mrlabel");
 	}
 
-	if((total == 0) && !possessEquipment($item[Mega Gem]) && lovemeDone && in_hardcore() && (item_amount($item[Wet Stunt Nut Stew]) == 0) && ((internalQuestStatus("questL11Palindome") >= 3) || isGuildClass()) && !get_property("auto_bruteForcePalindome").to_boolean())
+	if((total == 0) && !possessEquipment($item[Mega Gem]) && lovemeDone && (in_hardcore() || get_property("auto_doWhiteys").to_boolean()) && (item_amount($item[Wet Stunt Nut Stew]) == 0) && ((internalQuestStatus("questL11Palindome") >= 3) || isGuildClass()) && !get_property("auto_bruteForcePalindome").to_boolean())
 	{
 		if(item_amount($item[Wet Stunt Nut Stew]) == 0)
 		{
 			equipBaseline();
 			if((item_amount($item[Bird Rib]) == 0) || (item_amount($item[Lion Oil]) == 0))
 			{
-				if(item_amount($item[white page]) > 0)
-				{
-					set_property("choiceAdventure940", 1);
-					if(item_amount($item[Bird Rib]) > 0)
-					{
-						set_property("choiceAdventure940", 2);
-					}
-
-					if(get_property("lastGuildStoreOpen").to_int() < my_ascensions())
-					{
-						auto_log_warning("This is probably no longer needed as of r16907. Please remove me", "blue");
-						auto_log_warning("Going to pretend we have unlocked the Guild because Mafia will assume we need to do that before going to Whitey's Grove and screw up us. We'll fix it afterwards.", "red");
-					}
-					backupSetting("lastGuildStoreOpen", my_ascensions());
-					string[int] pages;
-					pages[0] = "inv_use.php?pwd&which=3&whichitem=7555";
-					pages[1] = "choice.php?pwd&whichchoice=940&option=" + get_property("choiceAdventure940");
-					if(autoAdvBypass(0, pages, $location[Whitey\'s Grove], "")) {}
-					restoreSetting("lastGuildStoreOpen");
-					return true;
-				}
-				// +item is nice to get that food
-				bat_formBats();
-				auto_lostStomach(true);
-				auto_log_info("Off to the grove for some doofy food!", "blue");
-				autoAdv(1, $location[Whitey\'s Grove]);
+				doWhiteys();
 			}
 			else if(item_amount($item[Stunt Nuts]) == 0)
 			{
@@ -3041,16 +3060,6 @@ boolean L11_palindome()
 			}
 		}
 
-		if((item_amount($item[Bird Rib]) > 0) && (item_amount($item[Lion Oil]) > 0) && (item_amount($item[Wet Stew]) == 0))
-		{
-			autoCraft("cook", 1, $item[Bird Rib], $item[Lion Oil]);
-		}
-
-		if((item_amount($item[Stunt Nuts]) > 0) && (item_amount($item[Wet Stew]) > 0) && (item_amount($item[Wet Stunt Nut Stew]) == 0))
-		{
-			autoCraft("cook", 1, $item[wet stew], $item[stunt nuts]);
-		}
-
 		if(!possessEquipment($item[Mega Gem]))
 		{
 			if (equipped_amount($item[Talisman o\' Namsilat]) == 0)
@@ -3095,8 +3104,36 @@ boolean L11_palindome()
 			}
 			else
 			{
-				// ran out of things to do, brute force it
-				set_property("auto_bruteForcePalindome",true);
+				//After we get the photos
+				//First try wishing, then try Whitey's if we have enough +item, then brute force.
+				//If we hit this, we should only need to finish the L11 quest so it won't hurt to do everything in provideItem
+				//since we will need +item for tomb rats in ~15 turns anyway. Buffs from wishes should still be active
+				//since they are 30 turns from monkey paw wishes and 20 turns from pocket/genie wishes.
+				if (internalQuestStatus("questL11Palindome") > 2)
+				{
+					if(auto_monkeyPawWishesLeft() > 0)
+					{
+						foreach it in $items[Lion Oil, Bird Rib]
+						{
+							if(item_amount(it) > 0) continue;
+							auto_makeMonkeyPawWish(it);
+						}
+						if(item_amount($item[Lion Oil]) > 0 && item_amount($item[Bird Rib]) > 0)
+						{
+							return makeWetStuntNutStew();
+						}
+						return false; //wasn't able to make the stew
+					}
+					else if(provideItem(300, $location[Whitey's Grove], true, true) >= 300)
+					{
+						set_property("auto_doWhiteys", true);
+						return doWhiteys(); //Initial call to do Whitey's Grove
+					}
+					else
+					{
+						set_property("auto_bruteForcePalindome",true);
+					}
+				}
 			}
 		}
 		if((my_mp() > 60) || considerGrimstoneGolem(true))

--- a/RELEASE/scripts/autoscend/quests/level_11.ash
+++ b/RELEASE/scripts/autoscend/quests/level_11.ash
@@ -2972,7 +2972,7 @@ boolean L11_palindome()
 		return false;
 	}
 
-	if(item_amount($item[wet stunt nut stew]) == 0 && internalQuestStatus("questL11Palindome") > 3)
+	if(item_amount($item[wet stunt nut stew]) == 0 && internalQuestStatus("questL11Palindome") >= 3)
 	{
 		return makeWetStuntNutStew();
 	}

--- a/RELEASE/scripts/autoscend/quests/level_11.ash
+++ b/RELEASE/scripts/autoscend/quests/level_11.ash
@@ -3085,11 +3085,19 @@ boolean L11_palindome()
 	}
 	else
 	{
-		if(!in_hardcore() && pulls_remaining() == 0 && !isAboutToPowerlevel())
+		if(!in_hardcore() && pulls_remaining() == 0)
 		{
 			// used our pulls today before getting to palindrome. Delay until next day or run out of other stuff to do
-			auto_log_debug("Delaying palindrome. In a normal run and don't have enough pulls to create wet stunt nut stew.");
-			return false;
+			if(!isAboutToPowerlevel())
+			{
+				auto_log_debug("Delaying palindrome. In a normal run and don't have enough pulls to create wet stunt nut stew.");
+				return false;
+			}
+			else
+			{
+				// ran out of things to do, brute force it
+				set_property("auto_bruteForcePalindome",true);
+			}
 		}
 		if((my_mp() > 60) || considerGrimstoneGolem(true))
 		{

--- a/RELEASE/scripts/autoscend/quests/level_11.ash
+++ b/RELEASE/scripts/autoscend/quests/level_11.ash
@@ -3085,6 +3085,12 @@ boolean L11_palindome()
 	}
 	else
 	{
+		if(!in_hardcore() && pulls_remaining() == 0 && !isAboutToPowerlevel())
+		{
+			// used our pulls today before getting to palindrome. Delay until next day or run out of other stuff to do
+			auto_log_debug("Delaying palindrome. In a normal run and don't have enough pulls to create wet stunt nut stew.");
+			return false;
+		}
 		if((my_mp() > 60) || considerGrimstoneGolem(true))
 		{
 			handleBjornify($familiar[Grimstone Golem]);

